api/src/pcapi/core/offerers/models.py

```diff
@@ -1208,3 +1208,39 @@ class OffererStats(PcObject, Base, Model):
         server_default="{}",
         nullable=False,
     )
+
+
+class Address(PcObject, Base, Model):
+    __tablename__ = "address"
+    banId: str | None = sa.Column(sa.Text(), nullable=True, unique=True)
+    inseeCode: str | None = sa.Column(sa.Text(), nullable=False)
+    street: str = sa.Column(sa.Text(), nullable=False)
+    postalCode: str = sa.Column(sa.Text(), nullable=False)
+    city: str = sa.Column(sa.Text(), nullable=False)
+    country: str = sa.Column(sa.Text(), nullable=False)
+    latitude: float = sa.Column(sa.Numeric(8, 5), nullable=False)
+    longitude: float = sa.Column(sa.Numeric(8, 5), nullable=False)
+
+    __table_args__ = (
+        sa.Index("ix_unique_address_per_banid", "banId", unique=True, postgresql_where=banId.isnot(None)),
+        sa.Index(
+            "ix_unique_address_per_street_and_insee_code",
+            "street",
+            "inseeCode",
+            unique=True,
+            postgresql_where=banId.is_(None),
+        ),
+        sa.CheckConstraint('length("postalCode") = 5'),
+        sa.CheckConstraint('length("inseeCode") = 5'),
+        sa.CheckConstraint('length("city") <= 50'),
+        sa.CheckConstraint('length("country") <= 50'),
+    )
+
+
+class OffererAddress(PcObject, Base, Model):
+    __tablename__ = "offerer_address"
+    label: str = sa.Column(sa.Text(), nullable=False)
+    addressId = sa.Column(sa.BigInteger, sa.ForeignKey("address.id"), index=True)
+    address: sa.orm.Mapped["Address"] = sa.orm.relationship("Address", foreign_keys=[addressId])
+    offererId = sa.Column(sa.BigInteger, sa.ForeignKey("offerer.id"), index=True)
```

Comment:
Lors de la suppression d'une structure ayant un lien avec une adresse (dans les conditions dans lesquelles il est possible de supprimer une structure), je pressens une exception. Tu dois pouvoir tester avec le bouton « Supprimer la structure » depuis le BO, sur la page d'une structure n'ayant jamais eu de réservation.

En effet tu ne supprimes pas le lien dans `pcapi.core.offerers.api.delete_offerer`.
Peut-être la simple suppression en cascade suffirait-elle, si des conditions métier ne s'y opposent pas ?
```suggestion
    offererId = sa.Column(sa.BigInteger, sa.ForeignKey("offerer.id", ondelete="CASCADE"), index=True)
```

Je perçois qu'on ne supprime pas d'objet `Address` donc la cascade ne semble pas nécessaire sur `addressId`.